backend/app/instance_provider.go

```diff
@@ -40,23 +42,28 @@ type instanceProvider struct {
 	factory InstanceFactoryFunc
 }
 
-func (ip *instanceProvider) GetKey(pluginContext backend.PluginContext) (interface{}, error) {
+func (ip *instanceProvider) GetKey(ctx context.Context, pluginContext backend.PluginContext) (interface{}, error) {
```

Comment:
>can you remind me again why can't we add the tenantID to the PluginContext instead?

In the short term, it's a bit confusing to add new tenant related information when we already have orgID. In the future, this may be where this belongs. But at this early stage, passing this information via context is pretty trivial and gets us moving forward. Similarly, since we use `PluginContext` as part of the public API, it's confusing for consumers to know what to do with this field. 

Regardless of whether or not `context` is used for this in the future, it's still good practice to propagate this through the instance provider.

>but here there are no API or requests involved, which could be misleading.

But instance management is only used as part of the request so I think this makes sense here. Since all plugin information is passed in each request, we use that request and the contextual information to give us the right instance to work with.

For example:

```go
func (s *Service) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
	dsHandler, err := s.getDataSourceHandler(ctx, req.PluginContext)
	if err != nil {
		return nil, err
	}
	return dsHandler.QueryData(ctx, req)
}

func (s *Service) getDataSourceHandler(ctx, pluginCtx backend.PluginContext) (*sqleng.DataSourceHandler, error) {
	i, err := s.im.Get(ctx, pluginCtx)
	if err != nil {
		return nil, err
	}
	instance := i.(*sqleng.DataSourceHandler)
	return instance, nil
}
```


Kinda similar to what we're doing -> [Google showcases a demo using user IP info as part of context for example](https://go.dev/blog/context)

Reply:
> In the short term, it's a bit confusing to add new tenant related information when we already have orgID.

I don't think it's confusing to have a tenantID and orgID (just minimal explanations of both fields should be enough).

> Similarly, since we use PluginContext as part of the public API, it's confusing for consumers to know what to do with this field.

If it's optional and documented, I don't think that would be confusing (also, I don't think there are external consumers apart from us?).

> But instance management is only used as part of the request so I think this makes sense here. Since all plugin information is passed in each request, we use that request and the contextual information to give us the right instance to work with.

I slightly disagree with this. That `ctx` makes sense in `QueryData` doesn't mean that makes sense for `InstanceManager.Get`. We already have a blob of metadata in `PluginContext`, passing a `context` just to pass more metadata seems like weak reasoning to me.

In general, I don't think that just passing the tenantID justifies this breaking change that affects so many plugins if there is an alternative. In any case, I may be missing things and since this has been approved already feel free to go forward with it :pray: